lib/simplecov/exit_codes/maximum_coverage_drop_check.rb

```diff
@@ -36,14 +38,24 @@ def last_run
         @last_run = SimpleCov::LastRun.read
       end
 
-      def coverage_diff
-        raise "Trying to access coverage_diff although there is no last run" unless last_run
-
-        @coverage_diff ||= last_run[:result][:covered_percent] - covered_percent
+      def coverage_drop_violations
+        @coverage_drop_violations ||=
+          compute_coverage_drop_violations.select do |achieved|
+            achieved.fetch(:max_drop) < achieved.fetch(:drop_percent)
+          end
       end
 
-      def covered_percent
-        SimpleCov.round_coverage(result.covered_percent)
+      def compute_coverage_drop_violations
```

Comment:
these aren't the violations yet right, just the coverage_drops? So I'd probably drop the `violations`. Maybe call it `_data` as it includes all the data surrounding a coverage drop? Not sure :shrug: